eng/cake/dotnet.cake

```diff
@@ -156,7 +156,11 @@ Task("dotnet-templates")
                 var projectName = template.Key.Split(":")[0];
                 var templateName = template.Key.Split(":")[1];
 
-                var framework = string.IsNullOrWhiteSpace(TestTFM) ? "" : $"--framework {TestTFM}";
+                var framework = string.IsNullOrWhiteSpace(TestTFM)
+                    ? ""
+                    : TestTFM.StartsWith("net6")
+                        ? $"--Framework {TestTFM}" // this is just needed until we ship the next MAUI SR2
+                        ? $"--framework {TestTFM}";
```

Comment:
Since the casing of the arg charged in the SDK and the lower case "workaround" is not yet shipped in maui SR2, we need this condition. Once maui SR2 goes out, we can use the lowercase everywhere.